src/test/regress/sql/multi_extension.sql

```diff
@@ -78,6 +78,71 @@ ALTER EXTENSION citus UPDATE TO '6.1-15';
 ALTER EXTENSION citus UPDATE TO '6.1-16';
 ALTER EXTENSION citus UPDATE TO '6.1-17';
 ALTER EXTENSION citus UPDATE TO '6.2-1';
+
+-- Test non-distributed queries work even in version mismatch
+SET citus.enable_version_checks TO 'true';
+
+-- Test CREATE TABLE
+CREATE TABLE version_mismatch_table(column1 int);
+
+-- Test COPY
+\copy version_mismatch_table FROM STDIN;
+0
+1
+2
+3
+4
+\.
+
+-- Test INSERT
+INSERT INTO version_mismatch_table(column1) VALUES(5);
+ 
+-- Test SELECT
+SELECT * FROM version_mismatch_table ORDER BY column1;
+
+-- Test SELECT from pg_catalog
```

Comment:
\l also prints collation and ctype information which is likely to be change among different systems. (For example my laptop uses tr_TR.UTF-8 but our servers uses en_US.UTF-8). Since you said you have no blocking feedback, I am skipping this one.